src/coreclr/vm/typedesc.cpp

```diff
@@ -221,14 +221,20 @@ void TypeDesc::ConstructName(CorElementType kind,
 
     case ELEMENT_TYPE_VAR:
     case ELEMENT_TYPE_MVAR:
+    {
         if (kind == ELEMENT_TYPE_VAR)
         {
-            ssBuff.Printf(W("!%d"), rank);
+            ssBuff.Append(W('!'));
         }
         else
         {
-            ssBuff.Printf(W("!!%d"), rank);
+            ssBuff.Append(W("!!"));
         }
+
+        WCHAR buffer[MaxSigned32BitDecString + 1];
+        FormatInteger(buffer, ARRAY_SIZE(buffer), "%d", rank);
```

Comment:
A nit - looking at all the `FormatInteger` usages, they add quite a bit of churn to the code. I was wondering if adding `AppendInteger` method on the `SString` would be cleaner.

Reply:
I had the same idea. I want to do this as well. My concern was the format. We have x, u, and d and expressing those is cumbersome at present. I want to take a hard look at `SString` in the future, see https://github.com/dotnet/runtime/issues/76581 for another example of annoyance. I can do that now or continue to focus on removing the printf logic, which I'd prefer. Once that is done `SString` is going to get an overhaul and `AppendInteger` is on the list.

Reply:
Sounds good then.